### PR TITLE
export readopt of prefix_same_as_start

### PR DESCRIPTION
**Pull request by lishihai9017**
using prefix_same_as_start in prefix seek simply our logic in some seeking scenario

### CODE REVIEW

File: tests/test_iterator.rs [hunk 1]
```diff
@@ -223,39 +223,51 @@ fn test_total_order_seek() {
...
+    }
+    assert!(key_count == 4);
+
+    let mut ropts = ReadOptions::new();
+    ropts.set_prefix_same_as_start(true);
+    let mut iter = db.iter_opt(ropts);
+    iter.seek(SeekKey::Key(b"k1-0"));
+    let mut key_count = 0;
+    while iter.valid() {
+        // only iterator sst files and memtable that contain keys has the same prefix with b"k1"
```
Comment (andelf) [inline tests/test_iterator.rs]: s/with/as/
s/iterator/iterate/
s/memtable/memtables/
s/has/with/

File: tests/test_prefix_extractor.rs [hunk 2]
```diff
@@ -14,11 +14,11 @@
 use rocksdb::*;
 use tempdir::TempDir;
 
-struct FixedSuffixTransform {
+struct FixedPrefixTransform {
```
Comment (andelf) [inline tests/test_prefix_extractor.rs]: prefix struct?
why a field named suffix_len

File: tests/test_iterator.rs [hunk 3]
```diff
@@ -14,17 +14,17 @@
 use rocksdb::*;
 use tempdir::TempDir;
 
-struct FixedSuffixTransform {
-    pub suffix_len: usize,
+struct FixedPrefixTransform {
```
Comment (siddontang) [inline tests/test_iterator.rs]: do we not need suffix test now?

File: tests/test_iterator.rs [hunk 4]
```diff
@@ -223,39 +223,51 @@ fn test_total_order_seek() {
...
+    db.put_opt(b"k3-1", b"a", &wopts).unwrap();
+    db.put_opt(b"k3-2", b"b", &wopts).unwrap();
+    db.put_opt(b"k3-3", b"c", &wopts).unwrap();
 
     let mut iter = db.iter();
     iter.seek(SeekKey::Key(b"k1-0"));
     let mut key_count = 0;
     while iter.valid() {
-        // only iterator sst files and memtable that contain keys has the same prefix with b"k1-0".
+        assert_eq!(keys[key_count], iter.key());
```
Comment (zhangjinpeng1987) [inline tests/test_iterator.rs]: Please add comment for this case.

File: tests/test_iterator.rs (hunk 1 above)
Comment (zhangjinpeng1987) [inline tests/test_iterator.rs]: and become invalid as soon as encounter a key with prefix not the same as start key.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**